File: smash-ranks-android/app/src/main/java/com/garpr/android/adapters/PlayersAdapter.java
package com.garpr.android.adapters;

import android.content.Context;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;

import com.garpr.android.R;
import com.garpr.android.models.AbsPlayer;
import com.garpr.android.models.PlayersBundle;

public class PlayersAdapter extends BaseAdapter<AbsPlayer> {

    public PlayersAdapter(@NonNull final Context context) {
        super(context);
        setHasStableIds(true);
    }

    public PlayersAdapter(@NonNull final LayoutInflater layoutInflater) {
        super(layoutInflater);
        setHasStableIds(true);
    }

    @Override
    public long getItemId(final int position) {
        return getItem(position).hashCode();
    }

    @LayoutRes
    @Override
    public int getItemViewType(final int position) {
        return R.layout.item_player;
    }

    public void set(@Nullable final PlayersBundle bundle) {
        if (bundle != null && bundle.hasPlayers()) {
            set(bundle.getPlayers());
        } else {
            clear();
        }
    }

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/adapters/PlayersAdapter.kt
package com.garpr.android.adapters

import android.content.Context
import androidx.annotation.LayoutRes
import com.garpr.android.R
import com.garpr.android.data.models.AbsPlayer
import com.garpr.android.data.models.FullTournament
import com.garpr.android.data.models.PlayersBundle

class PlayersAdapter(context: Context) : BaseAdapter<AbsPlayer>(context) {

    init {
        setHasStableIds(true)
    }

    override fun getItemId(position: Int): Long {
        return getItem(position).hashCode().toLong()
    }

    @LayoutRes
    override fun getItemViewType(position: Int) = R.layout.item_player

    fun set(content: FullTournament?) {
        set(content?.players)
    }

    fun set(content: PlayersBundle?) {
        set(content?.players)
    }

}
